Dto branch

rel1c-hub opened the pull request:
Add BookDto.class as a response DTO, and CreateBookRequestDto.class as a request DTO
Add Mapstruct and mappers
Return and accept DTOs on the service layer
Add BookController with methods:
public List getAll()
public BookDto getBookById(Long id)
public BookDto createBook(CreateBookRequestDto bookDto)
Add missing methods in the BookService and BookRepository
Create and use EntityNotFoundException class
Remove CommandLineRunner bean

rel1c-hub committed:
feat : base created

Created classes and interfaces for further compliation.

rel1c-hub committed:
feat : dto service impl

Implemented dto's for project, added mapstruct to project. Starter custom exception is created.

rel1c-hub committed:
fix : CommandLineRunner deleted

rel1c-hub committed:
fix : checkStyle failure fixed

azzilian commented on src/main/java/mate/academy/bookstore/mapper/BookMapper.java:
```diff
@@ -0,0 +1,14 @@
+package mate.academy.bookstore.mapper;
+
+import mate.academy.bookstore.config.MapperConfig;
+import mate.academy.bookstore.dto.BookDto;
+import mate.academy.bookstore.dto.CreateBookRequestDto;
+import mate.academy.bookstore.model.Book;
+import org.mapstruct.Mapper;
```
please do check packagename

rel1c-hub replied:
Packaging lgtm, dunno what I could change with it

azzilian commented on src/main/java/mate/academy/bookstore/dto/BookDto.java:
```diff
@@ -0,0 +1,14 @@
+
+@Data
+public class BookDto {
+    private Long id;
+    private String title;
+    private String author;
+    private BigDecimal price;
+    private String description;
+    private String coverImage;
```
Why isbn is missed?

rel1c-hub replied:
my bad, resolved

azzilian reviewed (changes requested):
Please do check your imports I beliece i is better if your project will be in the root directory

rel1c-hub committed:
fix : added isbn field

karma-o commented on src/main/java/mate/academy/bookstore/model/Book.java:
```diff
@@ -7,10 +7,12 @@
-import lombok.Data;
+import lombok.Getter;
+import lombok.Setter;
 
 @Entity
-@Data
+@Getter
+@Setter
```
we should also not forget about equals, hashcode & toString

karma-o commented on src/main/java/mate/academy/bookstore/repository/impl/BookRepositoryImpl.java:
```diff
@@ -36,6 +37,14 @@ public Book save(Book book) {
+        try (EntityManager entityManager = entityManagerFactory.createEntityManager()) {
+            Book book = entityManager.find(Book.class, id);
+            return Optional.ofNullable(book);
+        }
```
let's add `Exception` catch block just in case

karma-o commented on src/main/resources/application.properties:
```diff
@@ -8,3 +8,4 @@ spring.datasource.driver-class-name=com.mysql.jdbc.Driver
 spring.jpa.hibernate.ddl-auto=create-drop
 spring.jpa.show-sql=true
 spring.jpa.properties.hibernate.dialect=org.hibernate.dialect.MySQLDialect
+server.port=8081
```
any reason to not use the standard one?

rel1c-hub replied:
I found this solution on the stack; when I start the app on 8080, I always get an error: "port is already in use."

karma-o replied:
that means that you already have some Spring boot application running on your PC that is using that port, thus not allowing your to to run it again, you probably have either booted this application as a usual run, and in debug at the same time, or you quit the IDEA without terminating the program, just restart your PC and the problem should be gone, and let's remove this line of code after that

karma-o commented on src/main/java/mate/academy/bookstore/serivce/impl/BookServiceImpl.java:
```diff
@@ -11,14 +15,26 @@
+
+    @Override
+    public BookDto findById(Long id) {
+        Book book = bookRepository.findById(id).orElseThrow(
+                () -> new EntityNotFoundException("Can't find book by id" + id)
+        );
+        return bookMapper.toDto(book);
```
a small but pretty suggestion 😉 
```suggestion
        return bookRepository.findById(id)
                .map(bookMapper::toDto)
                .orElseThrow(
                    () -> new EntityNotFoundException("Can't find book by id" + id));
```

karma-o reviewed (changes requested):
left a couple minor comments

rel1c-hub committed:
fix : minor changes for model and getById method

added lombok anotations for Book model.
added exception for findById method, whereas needed
